### ISSUE TITLE
Can't Catch  Authentication Exception With Hub on Android

### ISSUE THREAD

**Issue by oras** [aws-amplify/amplify-android#755]
Hi, how I can catch authentication exception on android?
Using android API 28
implementation 'com.amplifyframework:aws-analytics-pinpoint:1.1.1'
implementation 'com.amplifyframework:aws-api:1.1.1'
implementation 'com.amplifyframework:aws-auth-cognito:1.1.1'
implementation 'com.amplifyframework:aws-datastore:1.1.1'
implementation 'com.amplifyframework:aws-predictions:1.1.1'
implementation 'com.amplifyframework:aws-storage-s3:1.1.1'

Hub Technic doesn't work 

```
Amplify.Hub.subscribe(HubChannel.AUTH,
            hubEvent -> {
                if (hubEvent.getName().equals(InitializationStatus.SUCCEEDED.toString())) {
                    Log.i("AuthQuickstart", "Auth successfully initialized");
                } else if (hubEvent.getName().equals(InitializationStatus.FAILED.toString())){
                    Log.i("AuthQuickstart", "Auth failed to succeed");
                } else {
                    switch (AuthChannelEventName.valueOf(hubEvent.getName())) {
                        case SIGNED_IN:
                            Log.i("AuthQuickstart", "Auth just became signed in.");
                            break;
                        case SIGNED_OUT:
                            Log.i("AuthQuickstart", "Auth just became signed out.");
                            break;
                        case SESSION_EXPIRED:
                            Log.i("AuthQuickstart", "Auth session just expired.");
                            break;
                        default:
                            Log.w("AuthQuickstart", "Unhandled Auth Event: " + AuthChannelEventName.valueOf(hubEvent.getName()));
                            break;
                    }
                }
            }
    );

```
I used the following solution but not stable for updates:

```
Amplify.Auth.signIn(
                this.userName,
                password,
                result -> Log.i("AuthQuickstart", result.isSignInComplete() ? "Sign in 
                succeeded" : "Sign in not complete"),
                this::errorAuthenticationHandle

);
```

```
private void errorAuthenticationHandle(AuthException error)
{
    Log.e("AmplifyQuickstart", error.toString());

    switch (getErrorNumber(error))
    {
         // Server require password reset
         case RESET:
         // Some code
         break;

         case FAILED:
          // Some code
         break;

        }
 }
```

**Comment by TrekSoft** [comment]
Hi, yeah Hub doesn't report on exceptions - those should be handled in the error handlers on the call. And yes we're aware the error types returned are not ideal and are actually just about to release an improvement to that which will be stable across new updates of the framework.

**Comment by jpignata** [comment]
This will go out in our next release.

**Comment by TrekSoft** [comment]
Which in fact has just been released today! Update to version 1.3.1 to get proper error types you can check.
